kotlin-checks-test-sources/src/main/kotlin/checks/ReturnInFinallyCheckSample.kt

```diff
@@ -0,0 +1,225 @@
+                }
+            }
+        }
+
+        fun continueToOuterLabelInFinally() {
+            outer@ while (true) {
+                try {
+                    throw RuntimeException()
+                } finally {
+                    inner@ do {
+                        continue@outer // Noncompliant {{Remove this continue statement from this finally block.}}
+                    } while (true)
+                }
+            }
+        }
+    }
+
+    class Compliant {
+        fun breakInForInFinally() {
+            try {
+                throw RuntimeException()
+            } finally {
+                for (ignored in "Hello, World!".chars()) {
+                    break // Compliant
+                }
+            }
+        }
+
+        fun breakInDoWhileInFinally() {
+            try {
+                throw RuntimeException()
+            } finally {
+                do {
+                    break // Compliant
+                } while (true)
+            }
+        }
+
+        fun breakInWhileInFinally() {
+            try {
+                throw RuntimeException()
+            } finally {
+                while (true) {
+                    break // Compliant
+                }
+            }
+        }
+
+        fun breakToLabelInFinally() {
+            while (true) {
+                try {
+                    throw RuntimeException()
+                } finally {
+                    inner@ do {
+                        break@inner // Compliant
+                    } while (true)
+                }
+            }
+        }
+
+        fun continueInForInFinally() {
+            try {
+                throw RuntimeException()
+            } finally {
+                for (ignored in "Hello, World!".chars()) {
+                    continue // Compliant
+                }
+            }
+        }
+
+        fun continueInDoWhileInFinally() {
+            try {
+                throw RuntimeException()
+            } finally {
+                do {
+                    continue // Compliant
+                } while (true)
+            }
+        }
+
+        fun continueInWhileInFinally() {
+            try {
+                throw RuntimeException()
+            } finally {
+                while (true) {
+                    continue // Compliant
+                }
+            }
+        }
+
+        fun continueToInnerLabelInFinally() {
+            while (true) {
+                try {
+                    throw RuntimeException()
+                } finally {
+                    inner@ do {
+                        continue@inner // Compliant
+                    } while (true)
+                }
+            }
+        }
+
+        fun returnInLambdaInFinally() {
+            while (true) {
+                try {
+                    throw RuntimeException()
+                } finally {
+                    listOf(1, 2, 3).forEach {
+                        if (it == 0) return // Compliant
```

Comment:
Sorry, I haven't noticed this previously:

this should not be compliant, because `return` without a label will return from the closest named function (described with keyword `fun`) So return in such cases are noncompliant.

And such code should be compliant:

```
listOf(1, 2, 3).forEach (fun f(it: Int) {
                        if (it == 0) return // Compliant
                   })
```

or 

```
listOf(1, 2, 3).forEach {
                        if (it == 0) return@forEach // Compliant
                    }
```

Here you can find more: https://kotlinlang.org/docs/returns.html#return-at-labels